### PR TITLE
scripts: add --go-ldflags argument to operator-sdk build command

### LINKED ISSUE #1435: Flag to provide Go build arguments
## Feature Request

**Is your feature request related to a problem? Please describe.**

I want to include Git commit hash in the operator binary. It can be injected as a build-time value as described in this blog: https://blog.alexellis.io/inject-build-time-vars-golang/
This requires a way to specify Go build arguments in the `operator-sdk build` command.

**Describe the solution you'd like**

The flag could be similar to what is used in `up local` sub-command:  `operator-sdk up local --go-ldflags`. 

---

Earlier I have added this feature request as a comment here: https://github.com/operator-framework/operator-sdk/issues/1404#issuecomment-491261300

### PR DESCRIPTION
**Pull request by ZoltanOnody**
**Description of the change:**
Add --go-ldflags support to operator-sdk build command

**Motivation for the change:**
I need to provide build time arguments to my operator such as version. With this MR I am able to run:
```
operator-sdk build image-name --go-ldflags "-X package.Version=vX.Y.Z"
```

Fixes #1435

### CODE REVIEW

**Comment by ZoltanOnody** [comment]
Sure, I'll have a look at it.

**Comment by ZoltanOnody** [comment]
I generalized it & amended my commit, now we can provide any additional arguments. Now it works like:
```
operator-sdk build image-name --go-build-args "-ldflags -X=package.Version=vX.Y.Z"
```

**Comment by LiliC** [comment]
@ZoltanOnody thanks so much!

/ok-to-test

**Comment by LiliC** [comment]
You can ignore the marker test, it seems to be behaving oddly right now. But the e2e test seems to be failing as well.

**Comment by ZoltanOnody** [comment]
The test seems to be working locally:

```
+ kubectl create namespace test-memcached
namespace/test-memcached created
+ operator-sdk test local ./test/e2e --up-local --namespace=test-memcached
INFO[0000] Testing operator locally.                    
ok  	github.com/operator-framework/operator-sdk/test/test-framework/test/e2e	47.360s
INFO[0049] Local operator test successfully completed.  
```

In Travis I see:

```
+operator-sdk test local ./test/e2e --up-local --namespace=test-memcached
INFO[0000] Testing operator locally.                    
time="2019-06-21T17:22:13Z" level=info msg="Started local operator"
--- FAIL: TestMemcached (80.15s)
    --- FAIL: TestMemcached/memcached-group (0.00s)
        --- FAIL: TestMemcached/memcached-group/Cluster2 (60.02s)
            memcached_test.go:100: Initialized cluster resources
            wait_util.go:48: Operator is running locally; skip waitForDeployment
            client.go:57: resource type  with namespace/name (test-memcached/example-memcached) created
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            memcached_test.go:114: timed out waiting for the condition
            client.go:75: resource type  with namespace/name (test-memcached/example-memcached) successfully deleted
FAIL
```

I think is just some random error caused by network issues in Travis. Could you please restart the job https://travis-ci.org/operator-framework/operator-sdk/jobs/548782267?

**Comment by LiliC** [comment]
@ZoltanOnody retriggered that test. 
Can you also rebase to current master, the marker test should be fixed by doing that, thanks for your patience!

**Comment by ZoltanOnody** [comment]
I rebased my branched and forcepushed it. When you retriggered the job, it passed so the failing test was not caused by my changes. :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 - Scaffold a `pkg/apis/<group>/group.go` package file to avoid `go/build` errors when running Kubernetes code generators. ([#1401](https://github.com/operator-framework/operator-sdk/pull/1401))
 - Adds a new extra variable containing the unmodified CR spec for ansible based operators. [#1563](https://github.com/operator-framework/operator-sdk/pull/1563)
 - New flag `--repo` for subcommands [`new`](https://github.com/operator-framework/operator-sdk/blob/master/doc/sdk-cli-reference.md#new) and [`migrate`](https://github.com/operator-framework/operator-sdk/blob/master/doc/sdk-cli-reference.md#migrate) specifies the repository path to be used in Go source files generated by the SDK. This flag can only be used with [Go modules](https://github.com/golang/go/wiki/Modules). ([#1475](https://github.com/operator-framework/operator-sdk/pull/1475))
+- Adds `--go-build-args` flag to `operator-sdk build` for providing additional Go build arguments
```
Comment (joelanford) [inline CHANGELOG.md]:
```suggestion
- Adds `--go-build-args` flag to `operator-sdk build` for providing additional Go build arguments. ([#1582](https://github.com/operator-framework/operator-sdk/pull/1582))
```

**Comment by ZoltanOnody** [comment]
Is it OK to change it using "Commit suggestion" via Github (I've seen this feature for the first time), or should a squash my commits locally and force push?

BTW, do you have any ETA for the next release of operator-sdk?

**Comment by joelanford** [comment]
> Is it OK to change it using "Commit suggestion" via Github (I've seen this feature for the first time), or should a squash my commits locally and force push?

Yep. Using "commit suggestion" is totally fine. When one of the maintainers actually does the merge, GitHub automatically squahes everything for us, so you don't have to squash and force push.

> BTW, do you have any ETA for the next release of operator-sdk?

Probably later this week unless something comes up that forces us to push it back.

**Pull request merged**